File: scripts/sync_readme.py
#!/usr/bin/env python3
import pathlib
import shutil
import subprocess
import sys

# Set up logging
import logging
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')

root = pathlib.Path(__file__).resolve().parents[1]
root_readme = root / "README.md"
assets_readme = root / "src" / "assets" / "README.md"
root_readme_en = root / "README_EN.md"
assets_readme_en = root / "src" / "assets" / "README_EN.md"


def copy_and_stage(src: pathlib.Path, dest: pathlib.Path) -> None:
    if src.exists():
        try:
            shutil.copyfile(src, dest)
            logging.info(f"Copied {src} to {dest}")
        except Exception as e:
            logging.error(f"Failed to copy file: {e}")
            sys.exit(1)
        try:
            subprocess.run(["git", "add", str(dest)], check=True)
            logging.info(f"Added {dest} to git")
        except subprocess.CalledProcessError as e:
            logging.error(f"Failed to add file to git: {e}")
            sys.exit(1)
    else:
        logging.error(f"File {src} does not exist")
        sys.exit(1)


copy_and_stage(root_readme, assets_readme)
copy_and_stage(root_readme_en, assets_readme_en)

File: scripts/sync_readme.ts
import fs from 'fs';
import path from 'path';
import { execFileSync } from 'child_process';
import { fileURLToPath } from 'url';
import logger from '../src/utils/logger';

// Resolve paths
const __dirname = path.dirname(fileURLToPath(import.meta.url));
const root = path.resolve(__dirname, '..');
const rootReadme = path.join(root, 'README.md');
const assetsReadme = path.join(root, 'src', 'assets', 'README.md');
const rootReadmeEn = path.join(root, 'README_EN.md');
const assetsReadmeEn = path.join(root, 'src', 'assets', 'README_EN.md');

function copyAndStage(src: string, dest: string): void {
  if (fs.existsSync(src)) {
    try {
      fs.copyFileSync(src, dest);
      logger.log(`Copied ${src} to ${dest}`);
    } catch (err: any) {
      logger.error(`Failed to copy file: ${err.message}`);
      process.exit(1);
    }
    try {
      execFileSync('git', ['add', dest]);
      logger.log(`Added ${dest} to git`);
    } catch (err: any) {
      logger.error(`Failed to add file to git: ${err.message}`);
      process.exit(1);
    }
  } else {
    logger.error(`File ${src} does not exist`);
    process.exit(1);
  }
}

copyAndStage(rootReadme, assetsReadme);
copyAndStage(rootReadmeEn, assetsReadmeEn);
